packages/core/utils/src/config.js

```diff
@@ -16,56 +16,24 @@ export type ConfigOptions = {|
   parse?: boolean,
 |};
 
-export async function resolveConfig(
+export function resolveConfig(
   fs: FileSystem,
   filepath: FilePath,
   filenames: Array<FilePath>,
-  opts: ?ConfigOptions,
-  root: FilePath = path.parse(filepath).root,
-): Promise<FilePath | null> {
-  filepath = await fs.realpath(path.dirname(filepath));
-
-  // Don't traverse above the module root
-  if (path.basename(filepath) === 'node_modules') {
-    return null;
-  }
-
-  for (const filename of filenames) {
-    let file = path.join(filepath, filename);
-    if ((await fs.exists(file)) && (await fs.stat(file)).isFile()) {
-      return file;
-    }
-  }
-
-  if (filepath === root) {
-    return null;
-  }
-
-  return resolveConfig(fs, filepath, filenames, opts);
+): Promise<?FilePath> {
+  // TODO: realpath
+  return Promise.resolve(
+    fs.findAncestorFile(filenames, path.dirname(filepath)),
```

Comment:
Do we even need an async variant anymore?

Reply:
I figured we could keep the API the same even so, also in case we decide to bring back an async variant later...